includes/embeds/class-amp-core-block-handler.php

```diff
@@ -379,30 +381,52 @@ static function ( $matches ) use ( $modal_state_property, &$submenu_toggles_coun
 			return $block_content;
 		}
 
-		// Replace micromodal toggle logic with AMP state and set modal state property name based on its ID.
-		$block_content = preg_replace(
-			'/\sdata-micromodal-trigger="modal-\w+"/',
-			sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
-			$block_content
-		);
-
 		$block_content = preg_replace_callback(
 			'/(?<=<button)\s[^>]+/',
-			static function ( $matches ) use ( $modal_state_property ) {
+			static function ( $matches ) use ( $modal_state_property, $is_interactive_block ) {
 				$new_block_content = $matches[0];
 
 				// Skip submenu toggles.
 				if ( false !== strpos( $new_block_content, 'wp-block-navigation-submenu__toggle' ) ) {
 					return $new_block_content;
 				}
 
-				// Replace micromodal toggle logic bound with buttons with AMP state.
-				if ( false !== strpos( $new_block_content, ' data-micromodal-close' ) ) {
-					$new_block_content = str_replace(
-						' data-micromodal-close',
-						sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
-						$new_block_content
-					);
+				if ( $is_interactive_block ) {
+					// Replace `data-wp-on--click` with AMP state on submenu open button.
+					if ( false !== strpos( $new_block_content, 'wp-block-navigation__responsive-container-open' ) ) {
+						$new_block_content = preg_replace(
+							'/\sdata-wp-on--click="[^"]+"/',
+							sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
+							$new_block_content
+						);
+					}
+
+					// Replace `data-wp-on--click` with AMP state on submenu close button.
+					if ( false !== strpos( $new_block_content, 'wp-block-navigation__responsive-container-close' ) ) {
+						$new_block_content = preg_replace(
+							'/\sdata-wp-on--click="[^"]+"/',
+							sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
+							$new_block_content
+						);
+					}
+				} else {
+					// Replace micromodal toggle logic bound with buttons with AMP state to open the modal.
+					if ( false !== strpos( $new_block_content, ' data-micromodal-trigger' ) ) {
+						$new_block_content = preg_replace(
+							'/\sdata-micromodal-trigger="modal-\w+"/',
+							sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
+							$new_block_content
+						);
+					}
+
+					// Replace micromodal toggle logic bound with buttons with AMP state to close the modal.
+					if ( false !== strpos( $new_block_content, ' data-micromodal-close' ) ) {
+						$new_block_content = str_replace(
+							' data-micromodal-close',
+							sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
+							$new_block_content
+						);
+					}
```

Comment:
No need for the `if` since it'll just do the search again in the replacement below.

```suggestion
					$new_block_content = str_replace(
						' data-micromodal-close',
						sprintf( ' on="tap:AMP.setState({ %1$s: !%1$s })"', esc_attr( $modal_state_property ) ),
						$new_block_content
					);
```